preprocess.lua

```diff
@@ -32,6 +32,8 @@ cmd:option('-shuffle', 1, [[Shuffle data]])
 cmd:option('-seed', 3435, [[Random seed]])
 
 cmd:option('-report_every', 100000, [[Report status every this many sentences]])
+cmd:option('-log_file', '', [[Outputs logs to a file under this path instead of stdout]])
```

Comment:
I think we should apply the pattern introduced by @srush: defining a static function `declareOpts` in the Logger that declares its options.

See for example with the Translator:

* https://github.com/OpenNMT/OpenNMT/blob/master/onmt/translate/Translator.lua#L3
* https://github.com/OpenNMT/OpenNMT/blob/master/translate.lua#L20